specifyweb/specify/api_tests.py

```diff
@@ -688,11 +698,104 @@ def test_agent_address_replacement(self):
 
         # Assert that the api request ran successfully
         response = c.post(
-            f'/api/specify/agent/replace/{agent_2.id}/{agent_1.id}/',
-            data=[],
-            content_type='text/plain')
+            f'/api/specify/agent/replace/{agent_1.id}/',
+            data=json.dumps({
+                'old_record_ids': [agent_2.id],
+                'new_record_data': None
+            }),
+            content_type='application/json'
+        )
         self.assertEqual(response.status_code, 204)
         
         # Assert there is only one address the points to agent_1
         self.assertEqual(models.Address.objects.filter(agent_id=7).count(), 1)
         self.assertEqual(models.Address.objects.filter(agent_id=6).exists(), False)
+
+    def test_agent_address_multiple_replacement(self):
+        c = Client()
+        c.force_login(self.specifyuser)
+
+        # Create agents and a collector relationship
+        agent_1 = models.Agent.objects.create(
+            id=7,
+            agenttype=0,
+            firstname="agent",
+            lastname="007",
+            specifyuser=None)
+        agent_2 = models.Agent.objects.create(
+            id=6,
+            agenttype=0,
+            firstname="agent",
+            lastname="006",
+            specifyuser=None)
+        agent_3 = models.Agent.objects.create(
+            id=5,
+            agenttype=0,
+            firstname="agent",
+            lastname="005",
+            specifyuser=None)
+
+        # Create mock addresses
+        models.Address.objects.create(
+            id=1,
+            timestampcreated="22022-11-30 14:34:51.000",
```

Comment:
I'm not terribly familiar with this, but does Django properly convert this to a Date Time object, or leave it as a string? 
It may matter when we make Django time-aware, and it's good to have the data as closely representative to what it would be in the 'real-world' application